src/redis.c

```diff
@@ -355,39 +356,37 @@ static int redis_handle_query(redisContext *rh, redis_node_t *rn,
 
 static int redis_db_stats(char *node, char const *info_line) /* {{{ */
 {
-  unsigned char db;
-  unsigned char max_db = 16;
-  static char field_name[10];
-  static char db_id[3];
-  char *str;
-  static char buf[MAX_REDIS_VAL_SIZE];
-  value_t val;
-  int i;
-
-  for (db = 0; db < max_db; db++) {
-	ssnprintf(field_name, sizeof(field_name), "db%d:keys", db);
-
-	str = strstr(info_line, field_name);
-	if (str) {
-	  str += strlen(field_name) + 1; /* also skip the '=' */
-	  for (i = 0; (*str && (isdigit((unsigned char)*str) || *str == '.')); i++, str++)
-		buf[i] = *str;
-	  buf[i] = '\0';
-
-	  if (parse_value(buf, &val, DS_TYPE_GAUGE) == -1) {
-		WARNING("redis plugin: Unable to parse field `%s'.", field_name);
-		return (-1);
-	  }
-
-	  ssnprintf(db_id, sizeof(db_id), "%d", db);
-	  redis_submit (node, "records", db_id, val);
-	}
+  for (int db = 0; db < REDIS_DEF_DB_COUNT; db++) {
+    static char buf[MAX_REDIS_VAL_SIZE];
+    static char field_name[10];
+    static char db_id[3];
+    value_t val;
+    char *str;
+    int i;
+
+    ssnprintf(field_name, sizeof(field_name), "db%d:keys", db);
+
+    str = strstr(info_line, field_name);
+    if (!str)
+      continue;
+
+    str += strlen(field_name) + 1; /* also skip the '=' */
```

Comment:
This comment is only helpful if you know the format being parsed. Please add a comment to the function describing the expected input format, e.g.:

```
/* redis_db_stats parses and dispatches Redis database statistics,
 * currently the number of keys for each database.
 * info_line needs to have the following format:
 *   db0:keys=4,expires=0,avg_ttl=0
 */